test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -484,4 +543,56 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReadyNonblockingMode", func() {
 			util.ExpectAdmittedWorkloadsTotalMetric(prodClusterQ, 2)
 		})
 	})
+
+	var _ = ginkgo.Context("Requeuing timestamp set to Creation", func() {
+
+		ginkgo.BeforeEach(func() {
+			podsReadyTimeout = 3 * time.Second
+			requeuingTimestamp = config.CreationTimestamp
+		})
+
+		ginkgo.It("Should keep the evicted workload at the front of the queue", func() {
+			// Build a standalone cluster queue with just enough capacity for a single workload.
+			// (Avoid using prod/dev queues to avoid borrowing)
+			standaloneClusterQ := testing.MakeClusterQueue("standalone-cq").
+				ResourceGroup(*testing.MakeFlavorQuotas("default").Resource(corev1.ResourceCPU, "1").Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, standaloneClusterQ)).Should(gomega.Succeed())
+			defer func() {
+				gomega.Expect(util.DeleteClusterQueue(ctx, k8sClient, standaloneClusterQ)).Should(gomega.Succeed())
+			}()
```

Comment:
Should we perform this in `ginkgo.AfterEach` in this `Context` node instead of the `defer`?

Reply:
IIUC this is because the queues are declared only within the `It`, while the `AfterEach` is for all tests.
Alternatively, we could declare the variable for all tests in the context, and in afterEach delete in not nil, but I don't hold a view which one is better.

Reply:
I updated to use a `JustBeforeEach()` (cant use `BeforeEach()` b/c you need the `k8sClient`) and an `AfterEach()`.

Reply:
Ah, we set up k8client in the `JustBeforeEach()` node...
Ideally, we want to set up the k8sclient in the `BeforeAll()` or `BeforeEach()`. However, I think the investigation is out of scope in this PR.